Python/Tests/TestAdapterTests/TestExecutorTests.cs

```diff
@@ -209,6 +209,7 @@ public class TestExecutorTests {
                 AssertUtil.ContainsAtLeast(resultNames, expectedResult.TestCase.FullyQualifiedName);
                 var actualResult = recorder.Results.SingleOrDefault(tr => tr.TestCase.FullyQualifiedName == expectedResult.TestCase.FullyQualifiedName);
                 Assert.AreEqual(expectedResult.Outcome, actualResult.Outcome, expectedResult.TestCase.FullyQualifiedName + " had incorrect result");
+                Assert.IsTrue(actualResult.Duration >= TimeSpan.Zero);
```

Comment:
Turns out that the duration is not accurate enough when run in the unit tests. It seemed okay in my manual testing in VS, but in the unit tests I'm getting a duration that is slightly shorter than it should be (for example a test that sleeps 0.5s has a duration of 0.46s). It's probably the messaging overhead that is causing it (more overhead at the start than at the end).

My first attempt at fixing this involved calculating the duration on the Python side, and it looks like I may have to do it that way for best accuracy.